src/HttpHeader.cc

```diff
@@ -486,6 +495,17 @@ HttpHeader::parse(const char *header_start, size_t hdrLen, Http::ContentLengthIn
             return 0;
         }
 
+        if (lines > 1 || hasBareCr) {
+            const auto framingHeader = (e->id == Http::HdrType::CONTENT_LENGTH || e->id == Http::HdrType::TRANSFER_ENCODING);
+            if (framingHeader) {
+                debugs(55, warnOnError, "WARNING: " << (lines > 1 ? "obs-fold" : hasBareCr) << " seen in " << e->name << ": " << e->value);
```

Comment:
I would rewrite this code to replace `hasBareCr` with a `problem` or a similar general variable and set the value of that variable to "suspicious CR", "obs-fold", and "obs-fold with suspicious CR" as needed. It would simplify the code, improve diagnostics, and reduce code duplication. This particular debugs() line would become:

```suggestion
                debugs(55, warnOnError, "WARNING: " << problem << " seen in framing-sensitive " << e->name << ": " << e->value);
```

However, if you would rather keep problem-dedicated marks/variables, then let's at least reduce the duplication and simplify a little:

```suggestion
                debugs(55, warnOnError, "WARNING: " << (hasBareCr ? hasBareCr : "obs-fold") << " seen in framing-sensitive " << e->name << ": " << e->value);
```
I do not insist on this change.